pyccel/ast/core.py

```diff
@@ -2566,9 +2566,23 @@ def results(self):
 
     @property
     def body(self):
-        """ CodeBlock containing all the statements in the function """
+        """
+        CodeBlock containing all the statements in the function.
+
+        Returns a CodeBlock contains all the statements in the function.
+        """
         return self._body
 
+    @body.setter
+    def body(self, body):
+        if iterable(body):
+            body = CodeBlock(body)
+        elif not isinstance(body, CodeBlock):
```

Comment:
I don't think this should be an if/elif block. Shouldn't it be:
```suggestion
        if iterable(body):
            body = CodeBlock(body)
        if not isinstance(body, CodeBlock):
```
? As is you don't save the result if it is an iterable

Reply:
The bot is complaining about the line 2580.
Cause in : https://github.com/pyccel/pyccel/blob/8c3ba1f6b90f7d3d5b45a70ff3d7c76037dfa28c/pyccel/parser/semantic.py#L3731-L3731
the body passed is an iterable.
The condition should stay an if/elif cause it access the first condition, so the second condition is always True in this case.

Reply:
Ah, sorry I misread this. For some reason I thought we had:
```python
        if iterable(body):
            body = CodeBlock(body)
        elif not isinstance(body, CodeBlock):
            self._body.remove_user_node(self)
            self._body = body
            self._body.set_current_user_node(self)
```
In that case I agree. This is fine :+1: